ddtrace/internal/processor/stats.py

```diff
@@ -0,0 +1,259 @@
+    return span.name, service, resource, _type, int(status_code), synthetics
+
+
+class SpanStatsProcessorV06(PeriodicService, SpanProcessor):
+    """SpanProcessor for computing, collecting and submitting span metrics to the Datadog Agent."""
+
+    def __init__(self, agent_url, interval=None, timeout=1.0, retry_attempts=3):
+        # type: (str, Optional[float], float, int) -> None
+        if interval is None:
+            interval = float(os.getenv("_DD_TRACE_STATS_WRITER_INTERVAL") or 10.0)
+        super(SpanStatsProcessorV06, self).__init__(interval=interval)
+        self._agent_url = agent_url
+        self._timeout = timeout
+        # Have the bucket size match the interval in which flushes occur.
+        self._bucket_size_ns = int(interval * 1e9)  # type: int
+        self._buckets = defaultdict(
+            lambda: defaultdict(SpanAggrStats)
+        )  # type: DefaultDict[int, DefaultDict[SpanAggrKey, SpanAggrStats]]
+        self._endpoint = "/v0.6/stats"
+        self._headers = {
+            "Datadog-Meta-Lang": "python",
+            "Datadog-Meta-Tracer-Version": ddtrace.__version__,
+            "Content-Type": "application/msgpack",
+        }  # type: Dict[str, str]
+        self._lock = Lock()
+        self._enabled = True
+        self._retry_request = tenacity.Retrying(
+            # Use a Fibonacci policy with jitter, same as AgentWriter.
+            wait=tenacity.wait_random_exponential(
+                multiplier=0.618 * self.interval / (1.618 ** retry_attempts) / 2, exp_base=1.618
+            ),
+            stop=tenacity.stop_after_attempt(retry_attempts),
+            retry=tenacity.retry_if_exception_type((httplib.HTTPException, OSError, IOError)),
+        )
+        self.start()
+
+    def on_span_start(self, span):
+        # type: (Span) -> None
+        pass
+
+    def on_span_finish(self, span):
+        # type: (Span) -> None
+        if not self._enabled:
+            return
+
+        is_top_level = _is_top_level(span)
+        if not is_top_level and not _is_measured(span):
+            return
+
+        with self._lock:
+            # Align the span into the corresponding stats bucket
+            assert span.duration_ns is not None
+            span_end_ns = span.start_ns + span.duration_ns
+            bucket_time_ns = span_end_ns - (span_end_ns % self._bucket_size_ns)
+            aggr_key = _span_aggr_key(span)
+            stats = self._buckets[bucket_time_ns][aggr_key]
+
+            stats.hits += 1
+            stats.duration += span.duration_ns
+            if is_top_level:
+                stats.top_level_hits += 1
+            if span.error:
+                stats.errors += 1
+                stats.err_distribution.add(span.duration_ns)
+            else:
+                stats.ok_distribution.add(span.duration_ns)
+
+    @property
+    def _agent_endpoint(self):
+        # type: () -> str
+        return "%s%s" % (self._agent_url, self._endpoint)
```

Comment:
Is this a property because we expect that this value could potentially change at runtime?

Reply:
Hmm not at the moment, this is probably just a result of basing the writing logic off of `AgentWriter` which can have the endpoint be downgraded.

The code isn't necessary here so we can do without it :+1: